scm-webapp/src/main/java/sonia/scm/update/repository/PublicFlagUpdateStep.java

```diff
@@ -89,9 +89,11 @@ private void addRepositoryReadPermissionForAnonymousUser(V1RepositoryHelper.V1Re
       .filter(V1Repository::isPublic)
       .forEach(v1Repository -> {
         Repository v2Repository = repositoryDAO.get(v1Repository.getId());
-        LOG.info(String.format("Add RepositoryRole 'READ' to _anonymous user for repository: %s - %s/%s", v2Repository.getId(), v2Repository.getNamespace(), v2Repository.getName()));
-        v2Repository.addPermission(new RepositoryPermission(v2AnonymousUser.getId(), "READ", false));
-        repositoryDAO.modify(v2Repository);
+        if (v2Repository != null) {
+          LOG.info(String.format("Add RepositoryRole 'READ' to _anonymous user for repository: %s - %s/%s", v2Repository.getId(), v2Repository.getNamespace(), v2Repository.getName()));
+          v2Repository.addPermission(new RepositoryPermission(v2AnonymousUser.getId(), "READ", false));
+          repositoryDAO.modify(v2Repository);
+        }
```

Comment:
Shouldn't we at least log the case that there is no suitable v2 repository? With a warning?

Reply:
Yeah, a log would be good, I'll add one. But a warning? Deleting repositories on migration is a valid option; a warning should be more like "something went somewhat wrong", and this is not necessarily wrong.